Source/DafnyDriver/Legacy/SynchronousCliCompilation.cs

```diff
@@ -306,13 +306,14 @@ public class SynchronousCliCompilation : IDisposable {
           compiled = false;
         }
 
-        var failBecauseOfDiagnostics = dafnyProgram.Reporter.FailCompilation;
+        var failBecauseOfDiagnostics = dafnyProgram.Reporter.FailCompilationMessage;
         if (!verified) {
           exitValue = ExitValue.VERIFICATION_ERROR;
         } else if (!compiled) {
           exitValue = ExitValue.COMPILE_ERROR;
-        } else if (failBecauseOfDiagnostics) {
+        } else if (failBecauseOfDiagnostics != null) {
           exitValue = ExitValue.DAFNY_ERROR;
+          await options.OutputWriter.WriteLineAsync($"Returning exit code {exitValue} because {failBecauseOfDiagnostics}");
```

Comment:
I guess this case is not tested for now, but I'm fine with that if you're following up with the change to fail on warnings for `dafny verify` too.

Reply:
I will